model/SearchEngine/Driver/Elasticsearch/ElasticSearch.php

```diff
@@ -114,6 +114,78 @@ public function search(Query $query): SearchResult
         return $this->buildResultSet($this->client->search($query));
     }
 
+    /**
+     * @fixme Duplicated from QueryBuilder
+     *
+     *        The index name is held in the QueryBuilder,
+     *        we may need to make it public or move there the code that makes
+     *        use of this method.
+     */
+    private function getIndexByType(string $type): string
+    {
+        if (isset(QueryBuilder::STRUCTURE_TO_INDEX_MAP[$type])) {
+            return $this->prefixer->prefix(
+                QueryBuilder::STRUCTURE_TO_INDEX_MAP[$type]
+            );
+        }
+
+        return IndexerInterface::UNCLASSIFIEDS_DOCUMENTS_INDEX;
+    }
+
+    public function queryByDocumentId(string $id, $type): SearchResult // ResultSet
```

Comment:
TODO:
- [ ] Not used anymore (most probably it won't be needed at all after finishing the implementation): Rollback these changes